### ISSUE TITLE
Angular Single Page - ERROR Error: There are no query params available for parsing. 

### ISSUE THREAD

**Issue by weilinzung** [auth0-samples/auth0-angular-samples#193]
Using the login example from here: https://github.com/auth0-samples/auth0-angular-samples/tree/master/01-Login 

It has this error. 

```
ERROR Error: There are no query params available for parsing.
    at e.<anonymous> (auth0-spa-js.production.esm.js:15)
```

https://github.com/auth0-samples/auth0-angular-samples/blob/master/01-Login/src/app/auth/auth.service.ts

```
  handleRedirectCallback$ = this.auth0Client$.pipe(
    concatMap((client: Auth0Client) => from(client.handleRedirectCallback()))
  );

private handleAuthCallback() {
    // Call when app reloads after user logs in with Auth0
    const params = window.location.search;
    if (params.includes('code=') && params.includes('state=')) {
      let targetRoute: string; // Path to redirect to after login processsed
      const authComplete$ = this.handleRedirectCallback$.pipe(
        // Have client, now call method to handle auth callback redirect
        tap(cbRes => {
          // Get and set target redirect route from callback results
          targetRoute = cbRes.appState && cbRes.appState.target ? cbRes.appState.target : '/';
        }),
        concatMap(() => {
          // Redirect callback complete; get user and login status
          return combineLatest([
            this.getUser$(),
            this.isAuthenticated$
          ]);
        })
      );
      // Subscribe to authentication completion observable
      // Response will be an array of user and login status
      authComplete$.subscribe(([user, loggedIn]) => {
        // Redirect to target route after callback processing
        this.router.navigate([targetRoute]);
      });
    }
  }
```

Could someone update the example with the latest version `v1.10.0`? Currently, I have downgraded to `1.8.1`. Thanks!

**Comment by stevehobbsdev** [comment]
Thanks for raising - could be related to https://github.com/auth0/auth0-spa-js/issues/515, we'll investigate.

**Comment by stevehobbsdev** [comment]
@weilinzung I've downloaded the [Angular quickstart sample](https://github.com/auth0-samples/auth0-angular-samples), installed `@auth0/auth0-spa-js@latest` and ran the project, and I am not getting the issues you are seeing.

Are you saying that you upgrade to `v1.10.0` and it breaks, and the only way to log in is to downgrade to `1.8.1`?

We can update the sample but I'd like to dig into why this isn't working. If you can provide a link to your app for us to take a look, that would really help. Alternatively, please see if you can reproduce the issue in the [SPA SDK Playground app](Package exports for '/path_to_project/sample/auth0-spa-js/node_modules/nanoid' do not define a valid './non-secure' target)

**Comment by Ro3A** [comment]
@stevehobbsdev I'm running into this as well at work which has a bit more complex routing requirements than my app at home, which I've not run into this. 

I've tried version 1.8.1, and downgrading doesn't solve it for me. I'm available to screenshare if you've got time

**Comment by Ro3A** [comment]
@stevehobbsdev 

I was able to get this sorted so posting what I found in case it helps others.

This error is a result of authService being re-initialized due to the app reloading, except without a code / state query string params. 
- Check for route config issues, compare route config carefully against the sample app. 
- Check for other redirects that may be interfering. 

In my case, I found that if I hard coded ` this.router.navigate(["/dashboard"])`, the error went away which helped me realize it must be a routing setting on my side which led me to  ` onSameUrlNavigation: 'reload' ` in our routing config. Since I was attempting to load the page with no path, this caused the app to reload and AuthService to reinitialize.

```
 // if the user has been redirected, handle the redirect
    this.handleRedirectCallback$
      .pipe(
        tap(() => this.initializedSubject.next(true)),
        concatMap(cbRes =>
          this.isAuthenticated$
            .pipe(
              map(loggedIn => ({
                loggedIn,
                targetUrl: cbRes.appState && cbRes.appState.target ? cbRes.appState.target : '/',
              }))
            )
        )
      )
     //.subscribe(result => this.router.navigate(result.targetUrl));
      .subscribe(result => this.router.navigate(["/dashboard"]));
```

To solve the issue, we have to check to see if result.targetUrl === window.location.href. If they're the same, we need to direct to a different url. I'm sure there are probably other ways to solve it, but this seemed to be the path of least resistance

So the code now looks like this:

```
      .subscribe(result => {
        //Check to see if targetUrl is same as current location. If it is, redirect to some default page
        if (result.targetUrl === window.location.href)
          this.router.navigate(["/dashboard"]);

        this.router.navigate(result.targetUrl)
      });
```

**Comment by weilinzung** [comment]
> @stevehobbsdev
> 
> I was able to get this sorted so posting what I found in case it helps others.
> 
> This error is a result of authService being re-initialized due to the app reloading, except without a code / state query string params.
> 
> * Check for route config issues, compare route config carefully against the sample app.
>   -- If you're using  `onSameUrlNavigation: 'reload'`, make sure you aren't redirecting to the same page as this will cause re-initialization of the auth service
> * Check for other redirects that may be interfering (Other auth guards with redirects, etc)
> 
> In my case, there was a second auth guard that was performing a redirect and interfering with the initialization of auth service.

hmmm. can you show a sample for your fix?

**Comment by Ro3A** [comment]
@weilinzung Unfortunately not, but I outlined what to look for and how to determine to root cause. 

Place a break point in the constructor of the authService or even better, in the handleAuthCallback method.. you'll find that it gets called twice (or more). When it does get called again, it's going to be without the state & code query string parameters which will cause the "there are no query params available for parsing" message. 

In other words, you need to make sure you're app isn't redirecting during the auth process. And if it must, then you need to ensure the state & code query params are retained through the redirect.

**Comment by weilinzung** [comment]
@Ro3A thanks for the advice!

That is also strange that happen after the v1.8.1.

**Comment by weilinzung** [comment]
@stevehobbsdev So I found the issue, there is a conflict with:

```
  {
    path: '**',
    redirectTo: '/',
    pathMatch: 'full',
  },
```

**Comment by stevehobbsdev** [comment]
Thanks everyone for their comments.

I would like to highlight something that we have been working on which will totally replace these samples, which is a brand new [Auth0 SDK for Angular](https://github.com/auth0/auth0-angular). Instead of continuing to iterate on what is essentially a wrapper component inside the Angular SPA sample, we have extracted it into a new library in its own right, [just as we did for React](https://github.com/auth0/auth0-react).

So all of this wrapper code has been rewritten and packaged as a new library. We've also updated the [quickstart](https://auth0.com/docs/quickstart/spa/angular-next) and [the sample](https://github.com/auth0-samples/auth0-angular-samples/tree/update/new-sdk/Sample-01) to use this new SDK.

As this thread has continued for a while, with many reports of problems and struggles, I would be very keen to hear of anyone who tries out the new SDK and whether it solves your problem or makes your experience with using Auth0 and Angular together easier. If you try it out, the best place to provide feedback is the [issue tracker on the new repository](https://github.com/auth0/auth0-angular/issues).

I'll close this for now as we'll not be iterating on the sample code further, but I'm excited to hear feedback on the new bits!
